src/discussions/common/HoverCard.test.jsx

```diff
@@ -117,69 +122,58 @@ describe('HoverCard', () => {
         rendered_body: rawBody,
         raw_body: rawBody,
       })];
     });
-    axiosMock.onPost(commentsApiUrl)
-      .reply(({ data }) => {
-        const {
-          rawBody,
-          threadId,
-        } = camelCaseObject(JSON.parse(data));
-        return [200, Factory.build(
-          'comment',
-          {
-            rendered_body: rawBody,
-            raw_body: rawBody,
-            thread_id: threadId,
-          },
-        )];
-      });
-
-    executeThunk(fetchThreads(courseId), store.dispatch, store.getState);
-    mockAxiosReturnPagedComments();
-    mockAxiosReturnPagedCommentsResponses();
+    axiosMock.onPost(commentsApiUrl).reply(({ data }) => {
+      const { rawBody, threadId } = camelCaseObject(JSON.parse(data));
+      return [200, Factory.build('comment', {
+        rendered_body: rawBody,
+        raw_body: rawBody,
+        thread_id: threadId,
+      })];
+    });
+
+    await executeThunk(fetchThreads(courseId), store.dispatch, store.getState);
+    await mockAxiosReturnPagedComments();
+    await mockAxiosReturnPagedCommentsResponses();
   });
 
   test('it should have hover card on post', async () => {
-    renderComponent(discussionPostId);
+    await waitFor(() => renderComponent(discussionPostId));
     const post = screen.getByTestId('post-thread-1');
     expect(within(post).getByTestId('hover-card-thread-1')).toBeInTheDocument();
   });
 
   test('it should have hover card on comment', async () => {
-    renderComponent(discussionPostId);
+    await waitFor(() => renderComponent(discussionPostId));
     const comment = await waitFor(() => screen.findByTestId('comment-comment-1'));
     expect(within(comment).getByTestId('hover-card-comment-1')).toBeInTheDocument();
   });
 
   test('it should show add response, like, follow and actions menu for hovered post', async () => {
-    renderComponent(discussionPostId);
+    await waitFor(() => renderComponent(discussionPostId));
     const post = screen.getByTestId('post-thread-1');
     const view = within(post).getByTestId('hover-card-thread-1');
+
     expect(within(view).queryByRole('button', { name: /Add response/i })).toBeInTheDocument();
     expect(within(view).getByRole('button', { name: /like/i })).toBeInTheDocument();
     expect(within(view).queryByRole('button', { name: /follow/i })).toBeInTheDocument();
     expect(within(view).queryByRole('button', { name: /actions menu/i })).toBeInTheDocument();
   });
 
   test('it should show add comment, Endorse, like and actions menu Buttons for hovered comment', async () => {
-    renderComponent(questionPostId);
-    const comment = await waitFor(() => screen.findByTestId('comment-comment-3'));
-    const view = within(comment).getByTestId('hover-card-comment-3');
+    await waitFor(() => renderComponent(discussionPostId));
+    const comment = await waitFor(() => screen.findByTestId('comment-comment-1'));
+    const view = within(comment).getByTestId('hover-card-comment-1');
```

Comment:
instead of view it should be menu, just need to replace the view title with menu